src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -206,6 +210,116 @@ public CtClass apply(CtType targetType) {
+        // contract: Given a statement with a templated initializer, substituteStatement should
+        // return a new statement with the initializer replaced with the value bound to the template
+        // parameter
+
+        // arrange
+        Factory factory = createFactoryWithTemplates();
+        CtClass<?> targetClass = factory.Class().create("TargetClass");
+
+        String templateExecutableName = "executable";
+        int templateVariableIndex = 0;
+        String templateVariableName = "s";
+        String initializerToSubstitute = "My chosen initializer";
+        CtStatement expectedStatement = factory.createLocalVariable(
+                factory.Type().stringType(), templateVariableName, factory.createLiteral(initializerToSubstitute)
+        );
+
+        StatementWithTemplatedInitializer template = new StatementWithTemplatedInitializer();
+        template._initializer_ = factory.createLiteral(initializerToSubstitute);
+
+        // act
+        CtStatement substitutedStatement = Substitution.substituteStatement(
+                targetClass, template, templateVariableIndex, templateExecutableName
+        );
+
+        // assert
+        assertEquals(expectedStatement, substitutedStatement);
+    }
+
+    private static class StatementWithTemplatedInitializer extends ExtensionTemplate {
+        TemplateParameter<String> _initializer_;
+
+        public void executable() {
+            String s = _initializer_.S();
+        }
+    }
+
+    @Test
+    public void testSubstituteFieldDefaultExpressionWithTemplatedInitializer() {
+        // contract: Give an expression with a templated initializer, substituteFieldDefaultExpression should
+        // return a new expression with the initializer replaced with the value bound to the template parameter
+
+        // arrange
+        Factory factory = createFactoryWithTemplates();
+        CtType<?> targetType = factory.Class().create("TargetClass");
+
+        String initializerToSubstitute = "My chosen initializer";
+        String templateFieldName = "s";
+        CtExpression expectedExpression =  factory.createCodeSnippetExpression(initializerToSubstitute);
+
+        ExpressionWithTemplatedInitializer template = new ExpressionWithTemplatedInitializer();
+        template._initializer_ = factory.createCodeSnippetExpression(initializerToSubstitute);
+
+        // act
+        CtExpression<?> substitutedExpression = Substitution.substituteFieldDefaultExpression(
+                targetType, template, templateFieldName
+        );
+
+        // assert
+        assertEquals(expectedExpression, substitutedExpression);
+    }
+
+    private static class ExpressionWithTemplatedInitializer extends ExtensionTemplate {
+        @Parameter
+        CtExpression _initializer_ = new CtExpressionImpl() {
+            @Override
+            public void accept(CtVisitor visitor) { }
+        };
+        String s = (String) _initializer_.S();
```

Comment:
I'm honestly a little bit uncertain how one is supposed to use templated initializers for fields that aren't primitives, as one must put a placeholder into the `_initializer_` field or there's an NPE on initializing the class. Anyway, I'd probably clean it up like so:

```suggestion
        TemplateParameter<String> _initializer_ = new CtLiteralImpl<>();
        String s = _initializer_.S();
```

@monperrus is there a better way to do this? I don't see any intuitive way to use a template argument in a field initializer, other than this somewhat dirty workaround.